command/agent/config.go

```diff
@@ -1437,6 +1437,20 @@ func FixupCheckType(raw interface{}) error {
 		case "service_id":
 			rawMap["serviceid"] = v
 			delete(rawMap, k)
+		case "header":
+			vm, ok := v.(map[string]interface{})
```

Comment:
This seems like it should be returning errors if things aren't what we expect, otherwise we could silently skip these.